File: graphql-test/src/main/java/com/junyss/graphqltest/equipment/model/entity/Equipment.java
package com.junyss.graphqltest.equipment.model.entity;

import javax.persistence.Entity;
import javax.persistence.Id;

import com.junyss.graphqltest.equipment.model.dto.request.EquipmentRequestDto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Equipment {

	@Id
	private String equipmentId;
	private String usedBy;
	private Integer count;
	private String newOrUsed;

	public static Equipment toEntity (EquipmentRequestDto equipmentRequestDto) {
		return new Equipment(
			equipmentRequestDto.getEquipmentId(),
			equipmentRequestDto.getUsedBy(),
			equipmentRequestDto.getCount(),
		equipmentRequestDto.getNewOrUsed());
	}
}

File: type-script-nest-graph-ql-실습/src/app/equipment/model/entities/equipment.ts
import { Column, Entity, PrimaryColumn, Unique } from "typeorm";
import { EquipmentRequestDto } from "../dto/request/equipment-request.dto";

@Entity("equipment")
@Unique(["equipmentId"])
export class Equipment {
  @PrimaryColumn("varchar", { name: "equipment_id", length: 255, nullable: false, comment: "ID" })
  equipmentId!: string;

  @Column("varchar", { name: "used_by", length: 255, nullable: false, comment: "사용자" })
  usedBy!: string;

  @Column("int", { name: "count", nullable: false, comment: "수량" })
  count!: number;

  @Column("varchar", { name: "new_or_used", length: 10, nullable: false })
  newOrUsed!: string;

  static toEntity(equipmentRequestDto: EquipmentRequestDto): Equipment {
    const equipment = new Equipment();
    equipment.equipmentId = equipmentRequestDto.equipmentId;
    equipment.usedBy = equipmentRequestDto.usedBy;
    equipment.count = equipmentRequestDto.count;
    equipment.newOrUsed = equipmentRequestDto.newOrUsed;
    return equipment;
  }
}
